lib/monopay-ruby/invoices/advanced_invoice.rb

```diff
@@ -0,0 +1,51 @@
+require_relative 'simple_invoice'
+
+module MonopayRuby
+  module Invoices
+    class AdvancedInvoice < MonopayRuby::Invoices::SimpleInvoice
+      attr_reader :other_merchant_paymant_info_params, :other_params
+
+      def create(amount, destination: nil, reference: nil, other_merchant_paymant_info_params: {}, other_params: {})
+        @other_merchant_paymant_info_params = other_merchant_paymant_info_params
+        @other_params = other_params
+
+        super(amount, destination: destination, reference: reference)
+      end
+
+      # Request body required for Monobank API
+      #
+      # @return [Hash] request body
+      def request_body
+        # TODO: add "ccy" and another missing params
+        # TODO: remove nil valued params
+        request_body = {
+          amount: amount,
+          redirectUrl: redirect_url,
+          webHookUrl: webhook_url,
+          merchantPaymInfo: {
+            reference: reference,
+            destination: destination
+          }
+        }
+
+        request_body[:merchantPaymInfo] = request_body[:merchantPaymInfo].merge!(other_merchant_paymant_info_params) # add additional merchant_payment_info params if they are present
+        request_body.merge!(other_params)                             # add additional params if they are present
+
+        # TODO: add sum params into present additional params
+        if request_body[:merchantPaymInfo].has_key?(:basketOrder)
+          request_body[:merchantPaymInfo][:basketOrder].first[:sum] = convert_to_cents(amount)
```

Comment:
this is unsafe, cehck if first param present to prevent NilClass undefined method error